src/content/docs/en/guides/server-side-rendering.mdx

```diff
@@ -230,4 +230,70 @@ if (!product) {
\ No newline at end of file
+The function takes an [endpoint context](/en/reference/api-reference/#endpoint-context) and returns a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response). A powerful feature of SSR, API routes are able to securely execute code on the server. To learn more, see our [Endpoints Guide](/en/core-concepts/endpoints/#server-endpoints-api-routes).
+
+### Streaming
+
+Astro enables HTTP streaming for all pages that are not prerendered. This means that your page's content will be sent incrementally to the page as it resolves async functions and requests.
+
+You can `await` promises in the frontmatter of your page, or you can move those requests down into Astro components the page renders. The page will then send down the HTML it's capable of sending down without waiting for those async components, and then it will start rendering the components in the order they are found in your page's HTML one by one.
+
+For example, given a page that fetches some data:
+
+```astro title="src/pages/index.astro"
+---
+const personResponse = await fetch('https://randomuser.me/api/');
+const personData = await personResponse.json();
+const randomPerson = personData.results[0];
+const factResponse = await fetch('https://catfact.ninja/fact');
+const factData = await factResponse.json();
+---
+<html>
+  <head>
+    <title>A name and a fact</title>
+  </head>
+  <body>
+    <p>{randomPerson.name.first}</p>
+    <p>{factData.fact}</p>
+  </body>
+</html>
+```
+
+We can move the fetch call into a component to allow the page to send down the shell of the markup (the html, head, and body) before the fetch request is done loading:
+
+```astro title="src/components/RandomName.astro"
+---
+const personResponse = await fetch('https://randomuser.me/api/');
+const personData = await personResponse.json();
+const randomPerson = personData.results[0];
+---
+<p>{randomPerson.name.first}</p>
+```
+
+```astro title="src/components/RandomFact.astro"
+---
+const factResponse = await fetch('https://catfact.ninja/fact');
+const factData = await factResponse.json();
+---
+<p>{factData.fact}</p>
+```
+
+And update our page to stream in the component instead:
+
+```astro title="src/pages/index.astro" {1, 9}
+---
+import RandomName from '../components/RandomName.astro'
+import RandomFact from '../components/RandomFact.astro'
+---
+<html>
+  <head>
+    <title>A name and a fact</title>
+  </head>
+  <body>
+    <RandomName />
+    <RandomFact />
+  </body>
+</html>
+```
+
+Now the page will stream in the HTML around the async components first, and then resolve each component one by one, streaming the resulting HTML.
```

Comment:
``````suggestion
To take advantage of Astro's HTML streaming, move the fetch calls into separate components included in the template. This allows the page to send down the shell of the markup (the `<html>`, `<head>`, and `<body>`) before the fetch request is done loading:

```astro title="src/pages/index.astro" {1, 9}
---
import RandomName from '../components/RandomName.astro'
import RandomFact from '../components/RandomFact.astro'
---
<html>
  <head>
    <title>A name and a fact</title>
  </head>
  <body>
    <RandomName />
    <RandomFact />
  </body>
</html>
```
Now the page will stream in the HTML around the async components first, and then resolve each component (shown below) one by one, streaming the resulting HTML.

```astro title="src/components/RandomName.astro"
---
const personResponse = await fetch('https://randomuser.me/api/');
const personData = await personResponse.json();
const randomPerson = personData.results[0];
---
<p>{randomPerson.name.first}</p>
```

```astro title="src/components/RandomFact.astro"
---
const factResponse = await fetch('https://catfact.ninja/fact');
const factData = await factResponse.json();
---
<p>{factData.fact}</p>
```
``````
I like what you've both done with the example here! (I'm half wondering whether this is now a recipe, but...)

I'm not enen *entirely* convinced we need to show the code for the individual components here, to be honest! (That's why my mind jumped to recipe.) What I did here was reorder to show essentially what we're trying to demonstrate: the comparison of what a page looks like when its fetch requests are in frontmatter (causing delays) and what a page looks like when streaming async components can happen. That *might* be all that's necessary here, to get the point across! Not sure, but take a look at how this page now reads with the subtle reordering and see what you think!

Reply:
@sarah11918 these edits are amazing!

I'll wait on the platform team's review to make sure we've got this right before committing to it.